File: client/src/app/_models/finance/voucherAttachment.ts

export interface IVoucherAttachment {
    id: number;
    voucherId: number;
    attachmentSizeInBytes: number;
    fileName: string;
    url: string;
    dateUploaded: Date;
    uploadedByUsername: string;
}

export class VoucherAttachment implements IVoucherAttachment {
    id: number=0;
    voucherId: number=0;
    attachmentSizeInBytes: number=0;
    fileName: string='';
    url: string='';
    dateUploaded: Date=new Date();
    uploadedByUsername="";
}

File: api/Entities/Finance/VoucherAttachment.cs
namespace api.Entities.Finance
{
    public class VoucherAttachment: BaseEntity
    {
        public VoucherAttachment()
        {
        }


        public int FinanceVoucherId { get; set; }
        public int AttachmentSizeInBytes { get; set; }
        public string FileName { get; set; }
        public string Url { get; set; }
        public DateTime DateUploaded { get; set; }
        public string UploadedByUsername { get; set; }
        public FinanceVoucher Voucher {get; set;}
    }
}